src/components/chatting/ChatWindow.jsx

```diff
@@ -172,102 +195,89 @@ const ChatWindow = () => {
-                            <div className='underline'>Reply</div>
-                          </div>
-                        </div>
-                      </div>
+                      <ChatCard
+                        data={chat}
+                        chatType={"message"}
+                        handleChatReply={handleChatReply}
+                      />
                     );
-                  } else
+                  } else {
                     return (
-                      <div key={index} className='flex justify-start gap-2'>
-                        <div className='flex flex-col'>
-                          <div className='text-sm font-semibold'>
-                            {chat.userName}
-                          </div>
-                          <div className='bg-gray-100 rounded-lg p-4 text-sm break-words dark:bg-gray-800'>
-                            {chat.message}
-                          </div>
-                          <div className='flex items-center gap-2 text-xs text-gray-500 dark:text-gray-400'>
-                            <div>{chat.time}</div>
-                            <div className='underline'>Reply</div>
-                          </div>
-                        </div>
-                      </div>
+                      <ChatCard
+                        data={chat}
+                        chatType={"reply"}
+                        handleChatReply={handleChatReply}
+                      />
                     );
-                } else if (chat.type === 'action' && chat.action === 'join')
+                  }
+                } else if (chat.type === "action" && chat.action === "join")
                   return (
-                    <div className='flex w-full justify-center text-slate-400'>
+                    <div
+                      key={crypto.randomUUID()}
+                      className="flex w-full justify-center text-slate-400"
+                    >
                       <span>{chat.userName} just joined the rooom !!!</span>
                     </div>
                   );
-                else if (chat.type === 'action' && chat.action === 'disconnect')
+                else if (chat.type === "action" && chat.action === "disconnect")
                   return (
-                    <div className='flex w-full justify-center text-slate-400'>
+                    <div
+                      key={crypto.randomUUID()}
+                      className="flex w-full justify-center text-slate-400"
+                    >
                       <span>{chat.userName} just left the rooom !!!</span>
                     </div>
                   );
                 else return null;
               })}
             </div>
+            <div className="reply-chat-pannel top-[100vh]">
+              {Object.keys(chatReply).length > 0 ? (
+                <ReplyCard
+                  data={chatReply}
+                  closeCard={closeReplyCard}
+                  isReplyCard={true}
```

Comment:
We can remove this `isReplyCard` property as we can check this same condition with `closeCard` callback.